File: app/src/main/java/com/sevtinge/cemiuiler/module/securitycenter/beauty/BeautyLight.java
package com.sevtinge.cemiuiler.module.securitycenter.beauty;

import com.sevtinge.cemiuiler.module.base.BaseHook;
import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.XposedBridge;
import de.robv.android.xposed.XposedHelpers;
import de.robv.android.xposed.callbacks.XC_LoadPackage;

import java.io.File;

import static com.sevtinge.cemiuiler.utils.Helpers.getPackageVersionCode;

public class BeautyLight extends BaseHook {
    @Override
    public void init() {
        int appVersionCode = getPackageVersionCode(lpparam);
        if (appVersionCode == 40000749 || appVersionCode == 40000750) {
            //hookAllMethods("p5.f", "m", XC_MethodReplacement.returnConstant(true));
            hookAllMethods("p5.f", "G", XC_MethodReplacement.returnConstant(true));
        } else if (appVersionCode == 40000754 || (appVersionCode >= 40000771 && appVersionCode <= 40000779 && appVersionCode != 40000772 && appVersionCode != 40000774 && appVersionCode != 40000777)) {
            findAndHookMethod("com.miui.gamebooster.beauty.l", "j", new BaseHook.MethodHook() {
                @Override
                protected void before(XC_MethodHook.MethodHookParam param) throws Throwable {
                    param.setResult(true);
                }
            });
        } else if (appVersionCode == 40000772) {
            findAndHookMethod("n5.f", "G", new BaseHook.MethodHook() {
                @Override
                protected void before(XC_MethodHook.MethodHookParam param) throws Throwable {
                    param.setResult(true);
                }
            });
        } else if (appVersionCode == 40000774) {
            findAndHookMethod("n5.f", "H", new BaseHook.MethodHook() {
                @Override
                protected void before(XC_MethodHook.MethodHookParam param) throws Throwable {
                    param.setResult(true);
                }
            });
        } else if (appVersionCode == 40000777 || appVersionCode == 40000780) {
            findAndHookMethod("p5.f", "H", new BaseHook.MethodHook() {
                @Override
                protected void before(XC_MethodHook.MethodHookParam param) throws Throwable {
                    param.setResult(true);
                }
            });
        } else {
            findAndHookMethod("com.miui.gamebooster.utils.o", "c", XC_MethodReplacement.returnConstant(true));
        }
        /*findAndHookMethod("com.miui.gamebooster.utils.o", "c", new BaseHook.MethodHook() {
            @Override
            protected void before(XC_MethodHook.MethodHookParam param) throws Throwable {
                param.setResult(true);
            }
        });*/
    }

}

File: app/src/main/java/com/sevtinge/cemiuiler/module/securitycenter/beauty/BeautyLight.kt
package com.sevtinge.cemiuiler.module.securitycenter.beauty

import com.github.kyuubiran.ezxhelper.HookFactory.`-Static`.createHook
import com.github.kyuubiran.ezxhelper.finders.MethodFinder.`-Static`.methodFinder
import com.sevtinge.cemiuiler.module.base.BaseHook
import com.sevtinge.cemiuiler.module.securitycenter.SecurityCenterDexKit
import com.sevtinge.cemiuiler.utils.Helpers
import com.sevtinge.cemiuiler.utils.api.isPad
import java.util.Objects


object BeautyLight : BaseHook() {
    override fun init() {
        try {
            val appVersionCode = Helpers.getPackageVersionCode(lpparam)
            val result =
                Objects.requireNonNull(SecurityCenterDexKit.mSecurityCenterResultClassMap["BeautyLight"])
            for (descriptor in result) {
                val beautyUtils = descriptor.getClassInstance(lpparam.classLoader)
                beautyUtils.methodFinder().first {
                    if (!isPad()) {
                        when {
                            appVersionCode in 40000749..40000750 -> name == "G"
                            appVersionCode in 40000754..40000756 -> name == "j"
                            appVersionCode in 40000761..40000762 -> name == "k"
                            appVersionCode in 40000771..40000772 -> name == "G"
                            appVersionCode >= 40000774 -> name == "H"// 手机端截止到 7.8.5-20230515 版本
                            else -> name == "c" // 未混淆分类
                        }
                    } else {
                        when {
                            appVersionCode == 40010749 -> name == "G"
                            appVersionCode == 40010750 -> name == "j"
                            appVersionCode in 40010771..40010772 -> name == "G"
                            appVersionCode >= 40010774 -> name == "H"// 平板端截止到 7.8.5-20230511 版本
                            else -> name == "j" // 未混淆分类
                        }
                    }
                }.createHook {
                    returnConstant(true)
                }
            }
        } catch (e: Throwable) {
            e.printStackTrace()
        }
    }
}
